modules/dxp/apps/portal-workflow/portal-workflow-metrics-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/service/internal/search/index/test/InstanceWorkflowMetricsIndexerTest.java

```diff
@@ -115,8 +128,86 @@ public void testReindex() throws Exception {
 			workflowDefinition.getWorkflowDefinitionId());
 	}
 
+	@Test
+	public void testUpdateAssigneeUser() throws Exception {
+		KaleoInstance kaleoInstance = getKaleoInstance(addBlogsEntry());
+
+		retryAssertCount(
+			booleanQuery -> booleanQuery.addMustQueryClauses(
+				queries.nested(
+					"tasks",
+					queries.term("tasks.assigneeType", Role.class.getName()))),
+			1,
+			_instanceWorkflowMetricsIndexNameBuilder.getIndexName(
+				workflowDefinition.getCompanyId()),
+			"WorkflowMetricsInstanceType", "className",
+			kaleoInstance.getClassName(), "classPK", kaleoInstance.getClassPK(),
+			"companyId", kaleoInstance.getCompanyId(), "completed", false,
+			"deleted", false, "instanceId", kaleoInstance.getKaleoInstanceId(),
+			"processId", workflowDefinition.getWorkflowDefinitionId(),
+			"version", "1.0");
+
+		User user = UserTestUtil.addUser(
+			RandomTestUtil.randomString(
+				NumericStringRandomizerBumper.INSTANCE,
+				UniqueStringRandomizerBumper.INSTANCE),
+			LocaleUtil.getDefault(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(),
+			new long[] {TestPropsValues.getGroupId()});
+
+		List<KaleoTaskInstanceToken> kaleoTaskInstanceTokens =
+			_kaleoTaskInstanceTokenLocalService.getKaleoTaskInstanceTokens(
```

Comment:
Hey @MarinhoFeliphe Can we move this to the `assignKaleoTaskInstanceToken` method somehow?
Thx